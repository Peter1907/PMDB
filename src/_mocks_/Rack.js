import { Link } from 'react-router-dom';
import * as Scroll from 'react-scroll';
import s from './Rack.module.css';
import store from './store';

const ScrollLink = Scroll.Link;

export default function Rack({ type }) {
  const saveIcon = ('../assets/add.png');
  const starIcon = ('../assets/star.png');
  const Data1 = store.details;
  const Data2 = store.popMovies;
  const Data3 = store.popSeries;
  const options = {
    'similar': Data1.similars,
    'movies': Data2,
    'series': Data3,
  };
  const selection = options[type];

  const refresh = () => {
    setTimeout(() => {
      document.location.reload();
    }, 100)
  };

  return (
    <div className={s.rack}>
      {(selection)
        && <div id={`${type}-rows`} className={s.rows}>
          <div name={`${type}-row1`} className={s.row1}>
            {selection.slice(0, 4).map((item, id) => (
              <div key={id} className={s.item}>
                <div className={s.posterContainer}>
                  <img className={s.saveTag} src={saveIcon} alt="#" />
                  <img className={s.poster} src={item.image} alt="poster" />
                </div>
                <div className={s.info}>
                  <Link onClick={() => refresh()} to={`/item-details/${item.id}`}>
                    <h3 className={s.title}>{item.title}</h3>
                  </Link>
                  <div className={s.ratingContainer}>
                    <p className={s.rating}>
                      <img src={starIcon} className={s.star} alt="star shaped icon" />
                      {item.imDbRating}
                    </p>
                  </div>
                  <button className={s.btn} type="button">Add to Watchlist</button>
                </div>
              </div>
            ))}
          </div>
          <div name={`${type}-row2`} className={s.row2}>
            {selection.slice(4, 8).map((item, id) => (
              <div key={id} className={s.item}>
                <div className={s.posterContainer}>
                  <img className={s.saveTag} src={saveIcon} alt="#" />
                  <img className={s.poster} src={item.image} alt="poster" />
                </div>
                <div className={s.info}>
                  <Link onClick={() => refresh()} to={`/item-details/${item.id}`}>
                    <h3 className={s.title}>{item.title}</h3>
                  </Link>
                  <div className={s.ratingContainer}>
                    <p className={s.rating}>
                      <img src={starIcon} className={s.star} alt="star shaped icon" />
                      {item.imDbRating}
                    </p>
                  </div>
                  <button className={s.btn} type="button">Add to Watchlist</button>
                </div>
              </div>
            ))}
          </div>
          <div name={`${type}-row3`} className={s.row3}>
            {selection.slice(8, 12).map((item, id) => (
              <div key={id} className={s.item}>
                <div className={s.posterContainer}>
                  <img className={s.saveTag} src={saveIcon} alt="#" />
                  <img className={s.poster} src={item.image} alt="poster" />
                </div>
                <div className={s.info}>
                  <Link onClick={() => refresh()} to={`/item-details/${item.id}`}>
                    <h3 className={s.title}>{item.title}</h3>
                  </Link>
                  <div className={s.ratingContainer}>
                    <p className={s.rating}>
                      <img src={starIcon} className={s.star} alt="star shaped icon" />
                      {item.imDbRating}
                    </p>
                  </div>
                  <button className={s.btn} type="button">Add to Watchlist</button>
                </div>
              </div>
            ))}
          </div>
        </div>}
      <div className={s.dotsContainer}>
        <ScrollLink
          to={`${type}-row1`}
          containerId={`${type}-rows`}
          spy={false}
          smooth={true}
          offset={50}
          duration={300}
          className={s.dot}
        >
        </ScrollLink>
        <ScrollLink
          to={`${type}-row2`}
          containerId={`${type}-rows`}
          spy={false}
          smooth={true}
          offset={50}
          duration={300}
          className={s.dot}
        >
        </ScrollLink>
        <ScrollLink
          to={`${type}-row3`}
          containerId={`${type}-rows`}
          spy={false}
          smooth={true}
          offset={50}
          duration={300}
          className={s.dot}
        >
        </ScrollLink>
      </div>
    </div>
  );
}
