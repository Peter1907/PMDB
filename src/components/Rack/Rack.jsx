import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import * as Scroll from 'react-scroll';
import PropTypes from 'prop-types';
import s from './Rack.module.css';
import ImgPlaceholder from '../ImgPlaceholder/ImgPlaceholder';

const ScrollLink = Scroll.Link;

export default function Rack({ type }) {
  const saveIcon = '../assets/add.png';
  const starIcon = '../assets/star.png';
  const Data1 = useSelector((state) => state.details);
  const Data2 = useSelector((state) => state.popMovies);
  const Data3 = useSelector((state) => state.popSeries);
  const options = {
    similar: Data1.similars,
    movies: Data2,
    series: Data3,
  };
  const selection = options[type];

  const refresh = (url) => {
    window.location.replace(window.location.origin + url);
  };

  const activateDot = (e) => {
    const dots = e.target.parentNode.childNodes;
    dots.forEach((dot) => {
      dot.classList.remove(`${s.active}`);
    });
    e.target.classList.add(`${s.active}`);
  };

  return (
    <div className={s.rack} onLoad={() => Scroll.scroller.scrollTo(0, 0)}>
      {selection && (
        <div id={`${type}-rows`} className={s.rows}>
          <div name={`${type}-row1`} className={s.row1}>
            {selection.slice(0, 4).map((item, id) => (
              <div key={id} className={s.item}>
                <div className={s.posterContainer}>
                  <img className={s.saveTag} src={saveIcon} alt="#" />
                  <ImgPlaceholder key={id} alt="poster" src={item.image} orientation="v" />
                </div>
                <div className={s.info}>
                  <Link onClick={() => refresh(`/item-details/${item.id}`)} to={`/item-details/${item.id}`}>
                    <h3 className={s.title}>{item.title}</h3>
                  </Link>
                  <div className={s.ratingContainer}>
                    <p className={s.rating}>
                      <img src={starIcon} className={s.star} alt="star shaped icon" />
                      {item.imDbRating}
                    </p>
                  </div>
                  <button className={s.btn} type="button">
                    Add to Watchlist
                  </button>
                </div>
              </div>
            ))}
          </div>
          <div name={`${type}-row2`} className={s.row2}>
            {selection.slice(4, 8).map((item, id) => (
              <div key={id} className={s.item}>
                <div className={s.posterContainer}>
                  <img className={s.saveTag} src={saveIcon} alt="#" />
                  <ImgPlaceholder key={id} alt="poster" src={item.image} orientation="v" />
                </div>
                <div className={s.info}>
                  <Link onClick={() => refresh(`/item-details/${item.id}`)} to={`/item-details/${item.id}`}>
                    <h3 className={s.title}>{item.title}</h3>
                  </Link>
                  <div className={s.ratingContainer}>
                    <p className={s.rating}>
                      <img src={starIcon} className={s.star} alt="star shaped icon" />
                      {item.imDbRating}
                    </p>
                  </div>
                  <button className={s.btn} type="button">
                    Add to Watchlist
                  </button>
                </div>
              </div>
            ))}
          </div>
          <div name={`${type}-row3`} className={s.row3}>
            {selection.slice(8, 12).map((item, id) => (
              <div key={id} className={s.item}>
                <div className={s.posterContainer}>
                  <img className={s.saveTag} src={saveIcon} alt="#" />
                  <ImgPlaceholder key={id} alt="poster" src={item.image} orientation="v" />
                </div>
                <div className={s.info}>
                  <Link onClick={() => refresh(`/item-details/${item.id}`)} to={`/item-details/${item.id}`}>
                    <h3 className={s.title}>{item.title}</h3>
                  </Link>
                  <div className={s.ratingContainer}>
                    <p className={s.rating}>
                      <img src={starIcon} className={s.star} alt="star shaped icon" />
                      {item.imDbRating}
                    </p>
                  </div>
                  <button className={s.btn} type="button">
                    Add to Watchlist
                  </button>
                </div>
              </div>
            ))}
          </div>
        </div>
      )}
      <div className={s.dotsContainer}>
        <ScrollLink
          to={`${type}-row1`}
          containerId={`${type}-rows`}
          spy={false}
          smooth={true}
          offset={50}
          duration={300}
          className={`${s.dot} ${s.active}`}
          onClick={(e) => activateDot(e)}
        />
        <ScrollLink
          to={`${type}-row2`}
          containerId={`${type}-rows`}
          spy={false}
          smooth={true}
          offset={50}
          duration={300}
          className={s.dot}
          onClick={(e) => activateDot(e)}
        />
        <ScrollLink
          to={`${type}-row3`}
          containerId={`${type}-rows`}
          spy={false}
          smooth={true}
          offset={50}
          duration={300}
          className={s.dot}
          onClick={(e) => activateDot(e)}
        />
      </div>
    </div>
  );
}

Rack.propTypes = {
  type: PropTypes.string.isRequired,
};
