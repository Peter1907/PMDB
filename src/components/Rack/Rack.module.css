.rack {
  margin: auto;
  max-width: 75rem;
  position: relative;
}

.rows {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  height: 25.5rem;
  overflow: hidden;
  scroll-snap-type: y mandatory;
  scroll-behavior: smooth;
}

.row1,
.row2,
.row3 {
  display: flex;
  gap: 1rem;
  scroll-snap-align: start;
}

.item {
  height: 25rem;
  width: 13rem;
  flex-shrink: 0;
  position: relative;
}

.posterContainer {
  position: relative;
  border: 3px solid black;
  background-color: black;
}

.poster {
  width: 100%;
}

.saveTag {
  width: 2rem;
  position: absolute;
  top: 0;
  left: 0.3rem;
  opacity: 0.8;
  cursor: pointer;
}

.saveTag:hover {
  opacity: 1;
  box-shadow: inset 0 0 0 80px #ffd369;
}

.info {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 8.8rem;
  position: absolute;
  bottom: 0;
  background: linear-gradient(black, #222831, black);
}

a {
  text-decoration: none;
}

.title {
  font-family: 'Barlow Semi Condensed', sans-serif;
  margin: 0.5rem;
  font-weight: 600;
  color: #ffd369;
}

.ratingContainer {
  justify-self: flex-end;
  position: absolute;
  bottom: 3.5rem;
  right: 0.5rem;
  display: flex;
  align-items: center;
  justify-content: flex-start;
}

.rating {
  display: flex;
  gap: 0.3rem;
  align-items: center;
  margin: 0;
  font-weight: 700;
  color: #eee;
  font-family: 'Cairo', sans-serif;
}

.rating img {
  width: 20px;
}

.btn {
  margin: auto 0.5rem 0.5rem;
  background-color: rgba(238, 238, 238, 0.137);
  border: none;
  font-size: 16px;
  height: 2.7rem;
  font-weight: 700;
  color: #eee;
  cursor: pointer;
}

.btn:hover {
  background-color: rgba(238, 238, 238, 0.171);
  color: #ffd369;
}

.btn:active {
  transform: scale(0.98);
}

.dotsContainer {
  position: absolute;
  left: 56rem;
  top: 50%;
  transform: translate(0, -50%);
  display: flex;
  flex-direction: column;
  gap: 2rem;
}

.dot {
  width: 0.7rem;
  height: 0.7rem;
  border-radius: 50%;
  background-color: #eee;
  opacity: 0.6;
  transition: opacity 250ms ease;
  cursor: pointer;
}

.dot:hover {
  opacity: 1;
}

.dot.active {
  box-shadow: 0 0 0 2px #222831, 0 0 0 4px #ffd369;
  opacity: 1;
}

@media screen and (max-width: 950px) {
  .item:last-child {
    display: none;
  }

  .dotsContainer {
    left: 42rem;
  }
}

@media screen and (max-width: 768px) {
  .row1,
  .row2,
  .row3 {
    justify-content: center;
    flex-wrap: wrap;
    max-width: 30rem;
  }
  
  .rows {
    height: 51rem;
  }

  .dotsContainer {
    top: auto;
    left: 15rem;
    bottom: -3rem;
    transform: translate(-50%, 0);
    height: 2rem;
    flex-direction: row;
    align-items: center;
  }
}
