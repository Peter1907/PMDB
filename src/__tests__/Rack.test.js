import renderer from 'react-test-renderer';
import Rack from '../_mocks_/Rack';

it('renders correctly', () => {
  const tree = renderer
    .create(<Rack />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
