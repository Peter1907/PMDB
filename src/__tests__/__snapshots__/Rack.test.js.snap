// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rack"
>
  <div
    className="dotsContainer"
  >
    <a
      className="dot"
      onClick={[Function]}
    />
    <a
      className="dot"
      onClick={[Function]}
    />
    <a
      className="dot"
      onClick={[Function]}
    />
  </div>
</div>
`;
